### PR TITLE
[REF] spreadsheet: spreadsheet model refactor

### PR DESCRIPTION
**Pull request by yunchen4**
This pr refactors the current model structure of spreadsheet-related module. 

Data fields are integrated into an abstract model `spreadsheet.mixin` and all sub-models are inherited from it. 

This way we can group all decode/encode logics into one place.

task [3222572](https://www.odoo.com/web#id=3222572&cids=1&menu_id=4720&action=333&active_id=2328&model=project.task&view_type=form)

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/spreadsheet/models/spreadsheet.py
Comment (LucasLefevre) [inline addons/spreadsheet/models/spreadsheet.py:1]: rename this file to `spreadsheet_mixin.py`

File: addons/spreadsheet/models/spreadsheet_mixin.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import base64
+from odoo import api, fields, models
+from odoo.addons.spreadsheet.utils import empty_spreadsheet_data_base64
+
+class SpreadsheetMixin(models.AbstractModel):
+    _name = "spreadsheet.mixin"
+    _description = "Spreadsheet mixin"
+    _auto = False
+
+    spreadsheet_binary_data = fields.Binary(required=True, default=empty_spreadsheet_data_base64())
```
Comment (LucasLefevre) [inline addons/spreadsheet/models/spreadsheet_mixin.py:13]:
```suggestion
    spreadsheet_binary_data = fields.Binary(required=True, string="Spreadsheet file", default=empty_spreadsheet_data_base64())
```

**Review by LucasLefevre — COMMENTED** [review]
(I think the upgrade branch is not synced with the other branches. Try rebasing all three of them)

**Review by LucasLefevre — COMMENTED** [review]
robodoo r+

**Comment by LucasLefevre** [comment]
robodoo r+

File: addons/spreadsheet/models/spreadsheet_mixin.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    _description = "Spreadsheet mixin"
+    _auto = False
+
+    spreadsheet_binary_data = fields.Binary(required=True, string="Spreadsheet file", default=empty_spreadsheet_data_base64())
+    spreadsheet_data = fields.Text(compute='_compute_spreadsheet_data', inverse='_inverse_spreadsheet_data')
+    thumbnail = fields.Binary()
+
+    @api.depends("spreadsheet_binary_data")
+    def _compute_spreadsheet_data(self):
+        for spreadsheet in self.with_context(bin_size=False):
```
Comment (rrahir) [inline addons/spreadsheet/models/spreadsheet_mixin.py:19]: @LucasLefevre question comme ça: tu sais pourquoi le bin_size dans le contexte avait du sens dans ce compute et pas dans l'inverse?

Comment (LucasLefevre) [reply addons/spreadsheet/models/spreadsheet_mixin.py:19]: dans le compute on lit le binary field (et on veut le vrai contenu, pas la size) alors que dans l'inverse, on ne le lit pas, on l'écrit ?

**Pull request closed without merging**